src/plugins/bfetch/server/streaming/create_compressed_stream.ts

```diff
@@ -9,20 +9,63 @@
 import { promisify } from 'util';
 import { Observable } from 'rxjs';
 import { catchError, concatMap, finalize } from 'rxjs';
-import { Logger } from '@kbn/core/server';
+import { AnalyticsServiceStart, Logger } from '@kbn/core/server';
 import { Stream, PassThrough } from 'stream';
 import { constants, deflate } from 'zlib';
+import { reportPerformanceMetricEvent } from '@kbn/ebt-tools';
 
 const delimiter = '\n';
 const pDeflate = promisify(deflate);
 
-async function zipMessageToStream(output: PassThrough, message: string) {
+const BFETCH_SERVER_ENCODING_EVENT_TYPE = 'bfetch_server_encoding';
+interface ZipStreamMetric {
+  time: number;
+  messageSize: number;
+}
+
+class StreamMetricCollector {
+  private readonly _collector: ZipStreamMetric[] = [];
+  addMetric(time: number, messageSize: number) {
+    this._collector.push({ time, messageSize });
+  }
+  getEBTPerformanceMetricEvent() {
+    let totalTime = 0;
+    let totalMessageSize = 0;
+    for (let i = 0; i < this._collector.length; i++) {
+      totalTime += this._collector[i].time;
+      totalMessageSize += this._collector[i].messageSize;
+    }
+    return {
+      eventName: BFETCH_SERVER_ENCODING_EVENT_TYPE,
+      duration: totalTime,
+      key1: 'message_count',
+      value1: this._collector.length,
+      key2: 'total_byte_size',
+      value2: totalMessageSize,
+      key3: 'stream_type',
+      value3: 1, // 1 == 'compressed'. Can always include support for ndjson-type later (e.g. 2 == ndjson)
+    };
+  }
+}
+
+async function zipMessageToStream(
+  output: PassThrough,
+  message: string,
+  collector?: StreamMetricCollector
+) {
   return new Promise(async (resolve, reject) => {
     try {
+      const before = Date.now();
```

Comment:
`performance.now` should be a better fit here

Reply:
thx, done!